src/app/questionnaire-item/store/transform-response.ts

```diff
@@ -154,41 +164,48 @@ const getResponseItems: (
 ) => (items: QuestionnaireItem[]) => fhir.r4.QuestionnaireResponseItem[] = (
   getIsItemEnabled
 ) =>
-  R.chain((item) => {
-    const isEnabled = getIsItemEnabled(item);
-    const getIsEnabledForSubItems = isEnabled
-      ? getIsItemEnabled
-      : R.always(false);
-    const responseItem: fhir.r4.QuestionnaireResponseItem = {
-      linkId: item.linkId,
-      text: item.text,
-    };
-    const getItems = R.pipe(
-      R.values,
-      getResponseItems(getIsEnabledForSubItems)
-    );
-    if (item.type === 'group') {
-      return R.map(
-        ({ items }) => ({
+  R.pipe<
```

Comment:
I would rather type the R.filter calls than the pipe.

E. g.

```typescript
  R.pipe(
    R.filter<QuestionnaireItem, 'array'>(getIsItemEnabled),
    R.chain((item) => {
      const responseItem: fhir.r4.QuestionnaireResponseItem = {
        linkId: item.linkId,
        text: item.text,
      };
      const getItems = R.pipe(R.values, getResponseItems(getIsItemEnabled));
      if (item.type === 'group') {
        return R.pipe(
          R.map(({ items }) => ({
            ...responseItem,
            item: getItems(items),
          })),
          R.filter<fhir.r4.QuestionnaireResponseItem, 'array'>(
            ({ item: answerItem }) =>
              !R.isNil(answerItem) && !R.isEmpty(answerItem)
          )
        )(item.itemAnswerList);
      }
      const responseAnswer = getResponseAnswers(item, getIsItemEnabled);
      if (R.isEmpty(responseAnswer)) {
        return;
      }
      return [
        {
          ...responseItem,
          answer: responseAnswer,
        } as fhir.r4.QuestionnaireResponseItem,
      ];
    }),
    R.filter<fhir.r4.QuestionnaireResponseItem, 'array'>(isNotNil)
  );
```

Reply:
ah nice, passing the `'array'` did the trick.